// @modalnodes/mn-docker/bundles/mn-mn-docker.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@mn/mn-docker', ['exports', '@angular/core', '@angular/common'], factory) :
    (factory((global.mn = global.mn || {}, global.mn['mn-docker'] = {}),global.ng.core,global.ng.common));
}(this, (function (exports,core,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var MnDockerService = (function () {
        function MnDockerService() {
            this.fileName = 'assets/env.json';
            this.conf = {};
            console.log('DockerService creating');
            this.conf = mn_docker_data;
        }
        /**
         * @return {?}
         */
        MnDockerService.prototype.loadConfigs = /**
         * @return {?}
         */
            function () {
                console.log('DockerService created', this.conf);
            };
        /**
         * @param {?} variableName
         * @param {?=} def
         * @return {?}
         */
        MnDockerService.prototype.getEnv = /**
         * @param {?} variableName
         * @param {?=} def
         * @return {?}
         */
            function (variableName, def) {
                if (def === void 0) {
                    def = '';
                }
                var /** @type {?} */ ret = '';
                if (Object.keys(this.conf).indexOf(variableName) >= 0) {
                    ret = this.conf[variableName];
                }
                else {
                    ret = def;
                }
                return ret;
            };
        /**
         * @param {?} variableName
         * @param {?=} def
         * @return {?}
         */
        MnDockerService.prototype.getEnvAsArray = /**
         * @param {?} variableName
         * @param {?=} def
         * @return {?}
         */
            function (variableName, def) {
                if (def === void 0) {
                    def = '';
                }
                var /** @type {?} */ env = this.getEnv(variableName, def);
                return JSON.parse(env.replace(/'/g, '"'));
            };
        /**
         * @return {?}
         */
        MnDockerService.prototype.getEnvironment = /**
         * @return {?}
         */
            function () {
                return this.conf;
            };
        MnDockerService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        MnDockerService.ctorParameters = function () { return []; };
        return MnDockerService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /**
     * @param {?} dockerService
     * @return {?}
     */
    function init_app(dockerService) {
        this.dockerService = dockerService;
        /**
         * @return {?}
         */
        function ret() {
            dockerService.loadConfigs();
        }
        return ret;
    }
    var MnDockerModule = (function () {
        function MnDockerModule() {
        }
        MnDockerModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [],
                        providers: [
                            MnDockerService,
                            { provide: core.APP_INITIALIZER, useFactory: init_app, deps: [MnDockerService], multi: true },
                        ]
                    },] },
        ];
        /** @nocollapse */
        MnDockerModule.ctorParameters = function () { return []; };
        return MnDockerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.MnDockerService = MnDockerService;
    exports.init_app = init_app;
    exports.MnDockerModule = MnDockerModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
