// @modalnodes/mn-docker/esm2015/lib/mn-docker.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
export class MnDockerService {
    constructor() {
        this.fileName = 'assets/env.json';
        this.conf = {};
        console.log('DockerService creating');
        this.conf = mn_docker_data;
    }
    /**
     * @return {?}
     */
    loadConfigs() {
        console.log('DockerService created', this.conf);
    }
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    getEnv(variableName, def = '') {
        let /** @type {?} */ ret = '';
        if (Object.keys(this.conf).indexOf(variableName) >= 0) {
            ret = this.conf[variableName];
        }
        else {
            ret = def;
        }
        return ret;
    }
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    getEnvAsArray(variableName, def = '') {
        const /** @type {?} */ env = this.getEnv(variableName, def);
        return JSON.parse(env.replace(/'/g, '"'));
    }
    /**
     * @return {?}
     */
    getEnvironment() {
        return this.conf;
    }
}
MnDockerService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
MnDockerService.ctorParameters = () => [];
function MnDockerService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MnDockerService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MnDockerService.ctorParameters;
    /** @type {?} */
    MnDockerService.prototype.fileName;
    /** @type {?} */
    MnDockerService.prototype.conf;
}
