// @modalnodes/mn-docker/fesm2015/mn-mn-docker.js
import { Injectable, NgModule, APP_INITIALIZER } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class MnDockerService {
    constructor() {
        this.fileName = 'assets/env.json';
        this.conf = {};
        console.log('DockerService creating');
        this.conf = mn_docker_data;
    }
    /**
     * @return {?}
     */
    loadConfigs() {
        console.log('DockerService created', this.conf);
    }
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    getEnv(variableName, def = '') {
        let /** @type {?} */ ret = '';
        if (Object.keys(this.conf).indexOf(variableName) >= 0) {
            ret = this.conf[variableName];
        }
        else {
            ret = def;
        }
        return ret;
    }
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    getEnvAsArray(variableName, def = '') {
        const /** @type {?} */ env = this.getEnv(variableName, def);
        return JSON.parse(env.replace(/'/g, '"'));
    }
    /**
     * @return {?}
     */
    getEnvironment() {
        return this.conf;
    }
}
MnDockerService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
MnDockerService.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @param {?} dockerService
 * @return {?}
 */
function init_app(dockerService) {
    this.dockerService = dockerService;
    /**
     * @return {?}
     */
    function ret() {
        dockerService.loadConfigs();
    }
    return ret;
}
class MnDockerModule {
}
MnDockerModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [],
                providers: [
                    MnDockerService,
                    { provide: APP_INITIALIZER, useFactory: init_app, deps: [MnDockerService], multi: true },
                ]
            },] },
];
/** @nocollapse */
MnDockerModule.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { MnDockerService, init_app, MnDockerModule };
