// @modalnodes/mn-docker/fesm5/mn-mn-docker.js
import { Injectable, NgModule, APP_INITIALIZER } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var MnDockerService = /** @class */ (function () {
    function MnDockerService() {
        this.fileName = 'assets/env.json';
        this.conf = {};
        console.log('DockerService creating');
        this.conf = mn_docker_data;
    }
    /**
     * @return {?}
     */
    MnDockerService.prototype.loadConfigs = /**
     * @return {?}
     */
    function () {
        console.log('DockerService created', this.conf);
    };
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    MnDockerService.prototype.getEnv = /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    function (variableName, def) {
        if (def === void 0) { def = ''; }
        var /** @type {?} */ ret = '';
        if (Object.keys(this.conf).indexOf(variableName) >= 0) {
            ret = this.conf[variableName];
        }
        else {
            ret = def;
        }
        return ret;
    };
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    MnDockerService.prototype.getEnvAsArray = /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    function (variableName, def) {
        if (def === void 0) { def = ''; }
        var /** @type {?} */ env = this.getEnv(variableName, def);
        return JSON.parse(env.replace(/'/g, '"'));
    };
    /**
     * @return {?}
     */
    MnDockerService.prototype.getEnvironment = /**
     * @return {?}
     */
    function () {
        return this.conf;
    };
    MnDockerService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    MnDockerService.ctorParameters = function () { return []; };
    return MnDockerService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @param {?} dockerService
 * @return {?}
 */
function init_app(dockerService) {
    this.dockerService = dockerService;
    /**
     * @return {?}
     */
    function ret() {
        dockerService.loadConfigs();
    }
    return ret;
}
var MnDockerModule = /** @class */ (function () {
    function MnDockerModule() {
    }
    MnDockerModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [],
                    providers: [
                        MnDockerService,
                        { provide: APP_INITIALIZER, useFactory: init_app, deps: [MnDockerService], multi: true },
                    ]
                },] },
    ];
    /** @nocollapse */
    MnDockerModule.ctorParameters = function () { return []; };
    return MnDockerModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { MnDockerService, init_app, MnDockerModule };
