// @modalnodes/mn-docker/esm5/lib/mn-docker.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
var MnDockerService = /** @class */ (function () {
    function MnDockerService() {
        this.fileName = 'assets/env.json';
        this.conf = {};
        console.log('DockerService creating');
        this.conf = mn_docker_data;
    }
    /**
     * @return {?}
     */
    MnDockerService.prototype.loadConfigs = /**
     * @return {?}
     */
    function () {
        console.log('DockerService created', this.conf);
    };
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    MnDockerService.prototype.getEnv = /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    function (variableName, def) {
        if (def === void 0) { def = ''; }
        var /** @type {?} */ ret = '';
        if (Object.keys(this.conf).indexOf(variableName) >= 0) {
            ret = this.conf[variableName];
        }
        else {
            ret = def;
        }
        return ret;
    };
    /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    MnDockerService.prototype.getEnvAsArray = /**
     * @param {?} variableName
     * @param {?=} def
     * @return {?}
     */
    function (variableName, def) {
        if (def === void 0) { def = ''; }
        var /** @type {?} */ env = this.getEnv(variableName, def);
        return JSON.parse(env.replace(/'/g, '"'));
    };
    /**
     * @return {?}
     */
    MnDockerService.prototype.getEnvironment = /**
     * @return {?}
     */
    function () {
        return this.conf;
    };
    MnDockerService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    MnDockerService.ctorParameters = function () { return []; };
    return MnDockerService;
}());
export { MnDockerService };
function MnDockerService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MnDockerService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MnDockerService.ctorParameters;
    /** @type {?} */
    MnDockerService.prototype.fileName;
    /** @type {?} */
    MnDockerService.prototype.conf;
}
