// @modalnodes/mn-docker/esm5/lib/mn-docker.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule, APP_INITIALIZER } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MnDockerService } from './mn-docker.service';
/**
 * @param {?} dockerService
 * @return {?}
 */
export function init_app(dockerService) {
    this.dockerService = dockerService;
    /**
     * @return {?}
     */
    function ret() {
        dockerService.loadConfigs();
    }
    return ret;
}
var MnDockerModule = /** @class */ (function () {
    function MnDockerModule() {
    }
    MnDockerModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [],
                    providers: [
                        MnDockerService,
                        { provide: APP_INITIALIZER, useFactory: init_app, deps: [MnDockerService], multi: true },
                    ]
                },] },
    ];
    /** @nocollapse */
    MnDockerModule.ctorParameters = function () { return []; };
    return MnDockerModule;
}());
export { MnDockerModule };
function MnDockerModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MnDockerModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MnDockerModule.ctorParameters;
}
